src/components/List.js

```diff
@@ -36,29 +45,47 @@ const List = ({ token }) => {
     <div>
       <h1>List</h1>
 
-      {docs.length === 0 && (
+      {docs.length === 0 ? (
         <section>
           <p>Your shopping list is currently empty.</p>
           <a href="/add-item">Add an Item</a>
         </section>
+      ) : (
+        <form>
+          <label for="search-bar">Filter Items</label>
+          <br />
+          <input
+            type="text"
+            name="search-bar"
+            id="search-bar"
+            placeholder="Start typing here..."
+            value={searchInput}
+            onChange={handleSearchChange}
+          />
+          <input type="reset" onClick={handleClear} />
+        </form>
       )}
 
       {errorMessage && <Error errorMessage={errorMessage} />}
 
       <ul style={{ listStyleType: 'none' }}>
         {docs &&
-          docs.map((doc) => (
-            <li key={doc.id}>
-              <input
-                type="checkbox"
-                name={doc.itemName}
-                id={doc.id}
-                onChange={handleCheckbox}
-                checked={checkPurchasedDate(doc.lastPurchased)}
-              />
-              {doc.itemName}
-            </li>
-          ))}
+          docs
+            .filter((doc) =>
+              doc.itemName.toLowerCase().includes(searchInput.toLowerCase()),
```

Comment:
Great idea to make the search not case sensitive! 🔤

Reply:
Agreed! It's a nice straightforward solution.

Reply:
Very nice! You might want to think about adding `.trim()`, which will trim the whitespace around either side of the value. That way if a user searches for ` apples` they will still find their `apples`.

Reply:
> Very nice! You might want to think about adding `.trim()`, which will trim the whitespace around either side of the value. That way if a user searches for ` apples` and accidentally types a space at the beginning or end, they will still find them.

So on the `seachInput.toLowerCase()`also chain on `.trim()`?

Reply:
Never mind on the question, added trim and it works 👍